test/test_optim.py

```diff
@@ -2290,6 +2290,34 @@ def avg_fn(p_avg, p, n_avg):
         for p_avg, p_swa in zip(averaged_params, averaged_dnn.parameters()):
             self.assertEqual(p_avg, p_swa)
 
+    def test_averaged_model_exponential_use_state_dict(self):
+        # Test AveragedModel with EMA as avg_fn and use_state_dict as True.
+        dnn = torch.nn.Sequential(
+            torch.nn.Conv2d(1, 5, kernel_size=3),
```

Comment:
Because your PR is specifically addressing issues with layers that have buffers it would be great to add a BatchNorm layer here.

Reply:
Makes sense, done.